File: java/json-transform/src/main/java/ai/bizone/jsontransform/functions/TransformerFunctionSplit.java
package ai.bizone.jsontransform.functions;

import ai.bizone.jsontransform.functions.common.*;
import ai.bizone.jsontransform.functions.common.*;

import java.util.Map;

public class TransformerFunctionSplit extends TransformerFunction {
    public TransformerFunctionSplit() {
        super(FunctionDescription.of(
                Map.of(
                        "delimiter", ArgumentType.of(ArgType.String).position(0).defaultValue(""),
                        "limit", ArgumentType.of(ArgType.Number).position(1).defaultValue(0)
                )
        ));
    }
    @Override
    public Object apply(FunctionContext context) {
        var str = context.getString(null);
        if (str == null) {
            return null;
        }
        return str.split(context.getString("delimiter"), context.getInteger("limit"));
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionSplit.ts
import TransformerFunction from "./common/TransformerFunction";
import { ArgType } from "./common/ArgType";
import FunctionContext from "./common/FunctionContext";
import { createComparator, getAsString, isNullOrUndefined } from "../JsonHelpers";
import CompareBy from "./common/CompareBy";
import JsonElementStreamer from "../JsonElementStreamer";
import javaSplit from "./utils/javaSplit";

class TransformerFunctionSplit extends TransformerFunction {
  constructor() {
    super({
      argsSets: [
        [
          { name: "delimiter", type: ArgType.String, defaultValue: "" },
          { name: "limit", type: ArgType.Number, defaultValue: 0 },
        ],
      ],
    });
  }

  override async apply(context: FunctionContext): Promise<any> {
    const str = await context.getString(null);
    if (str == null) {
      return null;
    }
    const delimiter = (await context.getString("delimiter")) ?? "";
    const limit = await context.getInteger("limit");
    return javaSplit(str, delimiter, limit);
  }
}

export default TransformerFunctionSplit;
